server/routes/updateGoal/updateGoalController.ts

```diff
@@ -60,6 +61,17 @@ export default class UpdateGoalController {
       return res.redirect(`/plan/${prisonNumber}/goals/${goalReference}/update#steps[${nextStepNumber - 1}][title]`)
     }
 
+    // Remove the desired step on the action delete step
+    if (updateStepForm.action.includes('delete-step-')) {
+      // Get the step reference inbetween the 2 characters [ ] from the action value
+      const stepReference = updateStepForm.action.match(/\[(.*?)\]/)[1]
```

Comment:
Couldn't think of a better way to find out which step to update than passing the reference through the button value